### DISCUSSION TITLE
Issues arising in ZMD workshop

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#8107] [Climatic]
I write following day 1.  Overall it went well.  I report on the gliches we found:

1. All machines are using the 64 bit version.  It seemed to install ok on each.
2. 2 machines had an immediate problem - they couldn't open the library dataset dialonds for tutorial 1.  Nor other files.  Francis had seen this problem before and has (apparently reported it.  He fixed it.  So all are now working ok.
3. In tutorial 1 we have an inconvenience when going to a new page - perhaps the last page with the diamonds.  Not serious, but the data window appears blank.  If you click then the data comes fine.  Why is it blank?  Can someone repeat it locally and make a small issue?
4. In tutorial 2 we imported the climatic data for Dodoma from the Excel file - as instructed.  We used the Describe > Summarise at the start and again after filtering to the data from 1958.  Sometimes the output window is incomplete.  You get to the bottom and this is not the bottom of the output.  (After doing the dialogue again it is ok)  Can someone get it "at home" to show.  Aha, I think I have found the problem.  It is perhaps a general one to check.  If I have output - say from Describe Summarise, and then I change the width of the output window, then it messes with the length, and hence with what is displayed.
5. We had an odd problem with the summarise dialogue, that I have not seen before.  Francis took a picture and will share.

**Comment by rdstern** [comment]
Day 2 of the workshop was - as usual, on the climatic menu.  We had 5 stations exported data from climsoft that we put into R-Instat and produced a single climatic data set.  All worked pretty well.  But:
a) `Make Date `3 variables, the focus doesn't seem to advance to the day variable.  More ambitiously perhaps would be to fill in the year, month, day if those are the names?  This happens nicely on the define climatic data, so perhaps there is some neat code to use there?  If so, then at least year, yr, and possibly annee - and allow for capitals.  Mon, and month, and even perhaps mois for the month, and day and jour for the day?  
b) In the Use Date dialogue 3 machines had the problem we keep meeting where they can't cope with the abbreviated names.  And I haven't checked with any other like decade that have a similar option.  Francis said he has fixed them, so I would like to know how, and will report, before expecting anything in the code. (He tells me he was able to type a line into the script window, and then it was fixed.  He will report and we can then consider whether that's the best we can do, or we incorporate that in the code.)
c) I used the Maximise for the graphs a lot, and it is a relief not to have to make so much use of the R-viewer.  However, I do like the way the graph can be maximised in the R viewer.  I think more work is planned on other aspects of the viewer and welcome this.
d) In the Check Data Climatic Boxplots I was keen to try the R viewer, to get full screen.  It doesn't seem to work for this dialogue.  I don't mind that omission if there is work on item c) soon.
e) We had a request to be able to clear the output window - just as we can clear the data frames?  If <ctrl> <A> worked, etc, then we could.  There is currently no right-click - I know Chris isn't keen on them.
f) The output sometimes gave a blank window - you had to scroll up - sometimes quite a long way to find the results.
g) In the define data as climatic it doesn't recognise TMPMIN and TMPMAX at tmin and tmax.  These are the names coming from climsoft, so perhaps they could be added.
h) @lloyddewit one message we didn't have.  Some machines stayed in R-Instat most of the day - including mine, and we have not had the memory reports that I have continued to have in the past month.  Perhaps it really is related to the calculator, because we haven't used that in the workshop and I used it regularly when I have had problems.

This list has grown, somewhat.  But that hides how well R-Instat has performed today.

**Reply by fran2or** [reply]
@rdstern ,  for Day 2 (b) this [link](https://stackoverflow.com/questions/5345132/sys-setlocale-request-to-set-locale-cannot-be-honored) helped me solve the problem. I went through the **Use Date**  dialog but didn't click **OK** once it was activated. Instead,  I click the **To Script** tab. Just before the generated script line from the **Use Date** dialog, I put the line of code `Sys.setlocale("LC_TIME", "C")` and it ran okay when all the lines of scripts were executed together.  After this,  **Use Date**  dialog now works okay without any error message on all the 3 machines which gave error messages earlier.

**Comment by rdstern** [comment]
I am leaping to day 5 where we have has some interesting problems.

1. @ChrisMarsh82 I thought the default number of variables was to be changed to 50?  It is on mine, because climatic often gives just over 30 variables.  But it still seems to be 30 on this installation, which created the usual discussion.  Please could it be 50 on the February release.

2. Some were annoyed that the summary dialogue (and climatic summary) fixes the variable names.  We have come to the same point in the new tabulation system, so adding a tab to the sub-dialogue here, to be able to specify the variable names, will be useful.

3. Francis machine and also one participant had the out-of-memory problem I am sometimes blessed with.  However, I had wondered if the calculator was the cause.  But here we had never used the calculator.  The participant had left her machine overnight, so it had been in R-Instat a long time.  I think something is checking and must be using memory steadily.

4. We hit a nasty problem with the PICSA rainfall graphs which Francis took a photo.  I couldn't resolve anything.  Reset didn't help, which usually did with this dialogue, but saving the data and starting again did resolve the issue.

5. We edited some individual cells in a filtered sheet, when the full grid had 700,000 rows.  The individual change takes quite a time to implement.  I wonder if there are improvements in the code that could make this process more efficient.  We should therefore also try with still longer data frames.

6. One participant was concerned that her maximised graphs were "shrunk" compared with those from the R-viewer.  We used the maximise feature a lot, (it is great in teaching)  so it will be good to add more features to it.  In particular could we resize the graphs, so they could be full width.  I assume they would have to be resizable, because full screen is not always desirable.
